### PR TITLE
Sparse ops: Support elementwise ops between sparse and diagonal matrix

### PR DESCRIPTION
**Pull request by isratnisa**
Resolves [Issue#4415](https://github.com/dmlc/dgl/issues/4415)

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/mock_sparse/elementwise_op_sp.py [hunk 1]
```diff
@@ -1,18 +1,20 @@
 """dgl elementwise operators for sparse matrix module."""
 import torch
 from .sp_matrix import SparseMatrix
+from .diag_matrix import DiagMatrix
+
+__all__ = ["add", "sub", "mul", "div", "rdiv", "power", "rpower"]
 
-__all__ = ['add', 'sub', 'mul', 'div', 'rdiv', 'power', 'rpower']
 
 def add(A, B):
```
Comment (jermainewang) [inline python/dgl/mock_sparse/elementwise_op_sp.py:9]: missing type annotation

File: tests/pytorch/mock_sparse/test_elementwise_op_sp.py [hunk 2]
```diff
@@ -36,12 +43,35 @@ def _test():
...
+            assert np.allclose(
+                op(A, A1).val, op(A.val, A1.val), rtol=1e-4, atol=1e-4
+            )
+
+    _test()
+
+
+@parametrize_idtype
+@parametrize_dtype
+# TODO (Israt): Add div operator or not. Div does not support matrices with
```
Comment (jermainewang) [inline tests/pytorch/mock_sparse/test_elementwise_op_sp.py:55]: You can have a separate test case for div.

File: tests/pytorch/mock_sparse/test_elementwise_op_sp.py [hunk 3]
```diff
@@ -70,8 +102,10 @@ def test_expose_op():
     dgl.mock_sparse.mul(A, A)
     dgl.mock_sparse.div(A, A)
 
-if __name__ == '__main__':
+
+if __name__ == "__main__":
```
Comment (jermainewang) [inline tests/pytorch/mock_sparse/test_elementwise_op_sp.py:106]: BTW, `__main__` is actually not required if you are using pytest to trigger the tests.

**Review by jermainewang — APPROVED** [review]
Mostly good. Some minor comments.

File: tests/pytorch/mock_sparse/test_elementwise_op_sp.py (hunk 3 above)
Comment (isratnisa) [reply tests/pytorch/mock_sparse/test_elementwise_op_sp.py]: Removed.

File: tests/pytorch/mock_sparse/test_elementwise_op_sp.py (hunk 2 above)
Comment (isratnisa) [reply tests/pytorch/mock_sparse/test_elementwise_op_sp.py]: Done.

File: python/dgl/mock_sparse/elementwise_op_sp.py (hunk 1 above)
Comment (isratnisa) [reply python/dgl/mock_sparse/elementwise_op_sp.py]: Done.

**Pull request merged**